llvm/lib/CodeGen/PeepholeOptimizer.cpp

```diff
@@ -1035,8 +1035,11 @@ bool PeepholeOptimizer::findNextSource(RegSubRegPair RegSubReg,
         return false;
 
       // Insert the Def -> Use entry for the recently found source.
-      ValueTrackerResult CurSrcRes = RewriteMap.lookup(CurSrcPair);
-      if (CurSrcRes.isValid()) {
+      auto [InsertPt, WasInserted] = RewriteMap.try_emplace(CurSrcPair, Res);
+
+      if (!WasInserted) {
+        ValueTrackerResult CurSrcRes = InsertPt->second;
```

Comment:
You might want to make this const reference because you are not modifying `ValueTrackerResult` within this `if` block.  Note that `ValueTrackerResult` contains a `SmallVector<RegSubRegPair, 2>`, so it's not completely cheap to copy.

```suggestion
        const ValueTrackerResult &CurSrcRes = InsertPt->second;
```